docs/source/data_analysis/HPC-module-SLEAP.md

```diff
@@ -0,0 +1,610 @@
+metrics.val.npz
+training_config.json
+training_log.csv
+```
+The SLEAP GUI on your local machine can be used to quickly evaluate the trained models.
+
+- Select *Predict* -> *Evaluation Metrics for Trained Models...*
+- Click on *Add Trained Models(s)* and select the folder containing the model(s) you want to evaluate.
+- You can view the basic metrics on the shown table or you can also view a more detailed report (including plots) by clicking *View Metrics*.
+
+## Model inference
+By inference, we mean using a trained model to predict the labels on new frames/videos.
+SLEAP provides the `sleap-track` command line utility for running inference
+on a single video or a folder of videos.
+
+Below is an example SLURM batch script that contains a `sleap-track` call.
+```{code-block} bash
+:caption: infer_slurm.sh
+:name: infer-slurm-sh
+:linenos:
+#!/bin/bash
+
+#SBATCH -J slp_infer # job name
+#SBATCH -p gpu # partition
+#SBATCH -N 1   # number of nodes
+#SBATCH --mem 64G # memory pool for all cores
+#SBATCH -n 16 # number of cores
+#SBATCH -t 0-02:00 # time (D-HH:MM)
+#SBATCH --gres gpu:rtx5000:1 # request 1 GPU (of a specific kind)
+#SBATCH -o slurm.%x.%N.%j.out # write STDOUT
+#SBATCH -e slurm.%x.%N.%j.err # write STDERR
+#SBATCH --mail-type=ALL
+#SBATCH --mail-user=user@domain.com
+
+# Load the SLEAP module
+module load SLEAP
+
+# Define directories for SLEAP project and exported training job
+SLP_DIR=/ceph/scratch/neuroinformatics-dropoff/SLEAP_HPC_test_data
+VIDEO_DIR=$SLP_DIR/videos
+SLP_JOB_NAME=labels.v001.slp.training_job
+SLP_JOB_DIR=$SLP_DIR/$SLP_JOB_NAME
+
+# Go to the job directory
+cd $SLP_JOB_DIR
+# Make a directory to store the predictions
+mkdir -p predictions
+
+# Run the inference command
+sleap-track $VIDEO_DIR/M708149_EPM_20200317_165049331-converted.mp4 \
+    -m $SLP_JOB_DIR/models/231010_164307.centroid/training_config.json \
+    -m $SLP_JOB_DIR/models/231010_164307.centered_instance/training_config.json \
+    --gpu auto \
+    --tracking.tracker simple \
+    --tracking.similarity centroid \
+    --tracking.post_connect_single_breaks 1 \
+    -o predictions/labels.v001.slp.predictions.slp \
+    --verbosity json \
+    --no-empty-frames
+```
+The script is very similar to the training script, with the following differences:
+- The time limit `-t` is set lower, since inference is normally faster than training. This will however depend on the size of the video and the number of models used.
+- The requested number of cores `n` and memory `--mem` are higher. This will depend on the requirements of the specific job you are running. It's best practice to try with a scaled-down version of your data first, to get an idea of the resources needed.
+- The requested GPU is of a specific kind (RTX 5000). This will again depend on the requirements of your job, as the different GPU kinds vary in GPU memory size and compute capabilities (see [wiki](https://wiki.ucl.ac.uk/display/SSC/CPU+and+GPU+Platform+architecture)).
```

Comment:
```suggestion
- The requested GPU is of a specific kind (RTX 5000). This will again depend on the requirements of your job, as the different GPU kinds vary in GPU memory size and compute capabilities (see [the SWC wiki](https://wiki.ucl.ac.uk/display/SSC/CPU+and+GPU+Platform+architecture)).
```